### PR TITLE
Fix litteral unicode code points

### PR DESCRIPTION
**Pull request by maar35**
* archive_html.sh:
  Archive old html files.

* parse_iffr_html.py:
  Print encoding to read the AZ file when in debug mode.
Fix the presence of litteral unicode code points.

* run_iffr_unit_tests.py:
  Add unit tests concerning the fix of litteral unicode code points.

* web_tools.py:
  Fix the presence of litteral unicode code points.

* FilmInfoDialogController.cs:
  Clean description from html tags, by interpreting them as attributed
text.

* FilmInfo.cs, FilmOutlinable.cs, FilmTableDelegate.cs,
ScreeningOutlinable.cs:
  Upgrade film description to attributed string.

### CODE REVIEW

File: FilmFestivalLoader/IFFR/run_iffr_unit_tests.py [hunk 1]
```diff
@@ -324,5 +328,51 @@ def new_screen_physical():
...
+    # Act.
+    with open(code_point_file, 'r', encoding='utf-8') as f:
+        code_point_text = f.read()
+    with open(clean_file, 'r', encoding='utf-8') as f:
+        clean_text = f.read()
+    parser.parse_props(code_point_text)
+    result_text = parser.description
+
+    # Assert.
+    return result_text, clean_text[35633:35897]
```
Comment (maar35) [inline FilmFestivalLoader/IFFR/run_iffr_unit_tests.py:374]: Three more unit test, thank you!

File: FilmFestivalLoader/Shared/web_tools.py [hunk 2]
```diff
@@ -26,6 +27,17 @@ def get_charset(file, byte_count=512):
...
 
 
+def fix_json(code_point_str):
+    """Replace HTML entity code points by the corresponding symbols.
+
+    @param code_point_str: String possibly containing not decoded HTML code points like '\u003c'.
+    @return: String with code points replaced by the corresponding symbols.
+    """
+
+    result_str = json.loads('"' + code_point_str + '"')
```
Comment (maar35) [inline FilmFestivalLoader/Shared/web_tools.py:37]: So it was that simple!
Thanks for finding this 😎

File: PresentScreenings/Entities/FilmInfo.cs [hunk 3]
```diff
@@ -86,7 +91,9 @@ public FilmInfo(int filmId, Film.FilmInfoStatus infoStatus, string description,
         {
             FilmId = filmId;
             InfoStatus = infoStatus;
-            FilmDescription = description;
+            RawFilmDescription = description;
+            AttributedFilmDescription = HtmlToAttributed(description);
+            FilmDescription = AttributedFilmDescription.Value;
```
Comment (maar35) [inline PresentScreenings/Entities/FilmInfo.cs:96]: Good solution!

File: PresentScreenings/Entities/FilmInfo.cs [hunk 4]
```diff
@@ -99,10 +106,10 @@ public override string ToString()
             var newLine = Environment.NewLine;
             var twoLines = newLine + newLine;
             var builder = new StringBuilder(Url + newLine);
-            if (FilmDescription != string.Empty)
+            if (RawFilmDescription != string.Empty)
             {
                 builder.AppendLine(newLine + "Description");
-                builder.AppendLine(FilmDescription);
+                builder.AppendLine(RawFilmDescription);
```
Comment (maar35) [inline PresentScreenings/Entities/FilmInfo.cs:112]: Please consider to add a comment to explain why `RawFilmDescription` is used instead of `FilmDescription`.

File: PresentScreenings/Entities/FilmInfo.cs [hunk 5]
```diff
@@ -320,6 +342,44 @@ private static Film.FilmInfoStatus StringToFilmInfoStatus(string name)
...
+        /// Search for HTML tags, remove them and apply a bold font
+        /// to the text range marked by the tags.
+        ///
+        /// It is presumed that:
+        ///     - Every start tag has a matching end tag
+        ///     - Tag ranges do not overlap
+        /// </summary>
+        /// <param name="attrText"> The attributed string to be searched from</param>
+        /// <param name="index">The index to start searching</param>
+        /// <returns>The new starting index, -1 when no tags are found.</returns>
```
Comment (maar35) [inline PresentScreenings/Entities/FilmInfo.cs:355]: Please add a user story to make this more general, including different attributes for different tags.

**Review by maar35 — COMMENTED** [review]
Great fix, super great extra!
I love those attributed strings❤️

File: FilmFestivalLoader/IFFR/run_iffr_unit_tests.py (hunk 1 above)
Comment (maar35) [reply FilmFestivalLoader/IFFR/run_iffr_unit_tests.py:374]: ☺️

File: PresentScreenings/Entities/FilmInfo.cs (hunk 3 above)
Comment (maar35) [reply PresentScreenings/Entities/FilmInfo.cs:96]: ☺️

File: PresentScreenings/Entities/FilmInfo.cs (hunk 4 above)
Comment (maar35) [reply PresentScreenings/Entities/FilmInfo.cs:112]: Done.

File: FilmFestivalLoader/Shared/web_tools.py (hunk 2 above)
Comment (maar35) [reply FilmFestivalLoader/Shared/web_tools.py:37]: ☺️

File: PresentScreenings/Entities/FilmInfo.cs (hunk 5 above)
Comment (maar35) [reply PresentScreenings/Entities/FilmInfo.cs:355]: Done.

**Review by maar35 — COMMENTED** [review]
Thanks!
